packages/vite-environment-provider-workerd/src/moduleUtils.ts

```diff
@@ -0,0 +1,71 @@
+import { init as initCjsModuleLexer, parse } from 'cjs-module-lexer';
+import { dirname, resolve } from 'node:path';
+import { readFile, stat } from 'node:fs/promises';
+
+export async function collectModuleInfo(
+  moduleCode: string,
+  moduleFilePath: string,
+): Promise<CjsModuleInfo | EsmModuleInfo> {
+  if (!isCommonJS(moduleCode)) {
+    return { isCommonJS: false };
+  }
+
+  await initCjsModuleLexer();
+
+  const namedExportsSet = new Set<string>();
+
+  const cjsLexerResult = parse(moduleCode);
+  cjsLexerResult.exports.forEach(namedExport =>
+    namedExportsSet.add(namedExport),
+  );
+  for (const reExport of cjsLexerResult.reexports) {
+    const reExportsPath = resolve(dirname(moduleFilePath), reExport);
+
+    const reExportsFileStat = await stat(reExportsPath);
+    if (!reExportsFileStat.isFile) {
+      throw new Error(
+        "Error: Found cjs re-export that doesn't point to a relative path",
+      );
+    }
+
+    const reExportsCode = await readFile(reExportsPath, 'utf8');
+    const reExportsInfo = await collectModuleInfo(reExportsCode, reExportsPath);
+
+    if (reExportsInfo.isCommonJS) {
+      reExportsInfo.namedExports.forEach(namedExport =>
+        namedExportsSet.add(namedExport),
+      );
+    }
+  }
+
+  const namedExports = [...namedExportsSet.values()];
```

Comment:
`.values()` shouldn't be needed here because Sets are already iterable.
```suggestion
  const namedExports = [...namedExportsSet];
```

Reply:
yes, my bad thanks! 👍